php/libraries/User.class.inc

```diff
@@ -525,6 +527,60 @@ class User extends UserPermissions
         $this->update($updateArray);
     }
 
+    /**
+     * Use the Pwned Passwords API to determine whether a user is trying
+     * to use a password that has been revealed in public data breaches.
+     *
+     * @param string $password The password to check for pwnage.
+     *
+     * @return bool Whether the password is known to be pwned.
+     */
+    function pwnedPassword(string $password): bool
```

Comment:
I think this should start with a config option check to disable this so that LORIS isn't unuseable at places that either don't want or can't allow access to external sites.

Reply:
Line 562 makes sure that the password change will continue if for some reason the pwnedPassword API is unreachable, including networking configs or outage of the site. 

Given that LORIS should still be usable with this check, would you still prefer an explicit config option? If so, should that be something in `config.xml` or?

Reply:
Without an explicit config check, it requires an attempt to contact the server to error out before it response to the user's request (which could take on the order of tens of seconds depending on the reason for the failure), even with that fallback. It's also possible that some users just won't want to use the external service, even though it's available.

I think it should go in the config tables, since we're trying to phase out config.xml